apps/ledger-live-desktop/src/renderer/webworkers/index.ts

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+
+const workers = new Map<string, Worker>();
+
+export async function initWorker(path: string, alias?: string) {
+  let workerPath = path;
+  if (!__DEV__) {
+    // In development, vite.js is smart enough to load the worker directly
+    // In production, we bundle the worker into a single file using esbuild
+    // The location of the file is ./[file].worker.js, so we need to rewrite the path
+    workerPath = workerPath.split("/").slice(-1)[0];
+    if (workerPath.endsWith(".ts")) {
+      workerPath = workerPath.replace(".ts", ".worker.js");
+    } else if (workerPath.endsWith(".js")) {
+      workerPath = workerPath.replace(".js", ".worker.js");
+    } else {
+      workerPath = workerPath + ".worker.js";
+    }
+  }
+  const worker = new Worker(workerPath, { type: "module" });
+  workers.set(alias ?? path, worker);
+  return worker;
+}
+
+export function getWorker(pathOrAlias: string) {
+  return workers.get(pathOrAlias);
+}
+
+export function disposeWorker(pathOrAlias: string) {
+  const worker = workers.get(pathOrAlias);
+  if (worker) {
+    worker.terminate();
+    workers.delete(pathOrAlias);
+  }
+}
+
+export function useWorker(workerPath: string, onMessage: (e: MessageEvent) => void): void;
+export function useWorker(
+  workerPath: string,
+  workerAlias: string,
+  onMessage: (e: MessageEvent) => void,
+): void;
+export function useWorker(
+  workerPath: string,
+  workerAliasOrOnMessage: string | ((e: MessageEvent) => void),
+  maybeOnMessage?: (e: MessageEvent) => void,
+): void {
```

Comment:
We might want to take a look at some open source solutions for this
I found https://useworker.js.org/ (https://github.com/alewin/useWorker/blob/master/packages/useWorker/src/useWorker.ts)
Maybe it's not as optimised for our use-case, but there is some part in our implementation with the `effectCounter` for exemple that would need more comments to explain why it's needed

Reply:
i'm going to remove this unused code because i didn't needed the hook part